### PR TITLE
feat(python,rust,cli): support left and right anti/semi joins from the SQL interface

### PR DESCRIPTION
**Pull request by alexander-beedie**
Adds support for `[LEFT|RIGHT] [ANTI|SEMI] JOIN` SQL syntax:
```python
import polars as pl
frames = {
  "tbl_a": pl.DataFrame({"a": [1,2,3], "b": [4,0,6], "c": ["w","y","z"]}),
  "tbl_b": pl.DataFrame({"a": [3,2,1], "b": [6,5,4], "c": ["x","y","z"]}),
  "tbl_c": pl.DataFrame({"c": ["w","y","z"], "d": [10.5,-50.0,25.5]}),
}
with pl.SQLContext(frames, eager_execution=True) as ctx:
  df = ctx.execute(
    """
    SELECT *
    FROM tbl_a
    RIGHT ANTI JOIN tbl_b USING (b)
    LEFT SEMI JOIN tbl_c USING (c)
    """
  )

# shape: (1, 3)
# ┌─────┬─────┬─────┐
# │ a   ┆ b   ┆ c   │
# │ --- ┆ --- ┆ --- │
# │ i64 ┆ i64 ┆ str │
# ╞═════╪═════╪═════╡
# │ 2   ┆ 5   ┆ y   │
# └─────┴─────┴─────┘
```
Also exposes dedicated `anti_join` and `semi_join` methods to Rust (every other join type is already available, but anti/semi were relegated to the generic `join` method ;)

### CODE REVIEW

**Comment by alexander-beedie** [comment]
Having issues getting the `semi_anti_join` feature gate activated in polars-sql; clearly not doing so at the appropriate level of indirection, but can't for the life of me seem to find the right place for it... I'm sure this is a _very_ easy one, so any pointers appreciated 🤣

File: crates/polars-sql/Cargo.toml [hunk 1]
```diff
@@ -26,3 +26,4 @@ json = ["polars-lazy/json"]
 default = []
 ipc = ["polars-lazy/ipc"]
 parquet = ["polars-lazy/parquet"]
+semi_anti_join = ["polars-lazy/semi_anti_join"]
```
Comment (ritchie46) [inline crates/polars-sql/Cargo.toml:29]: Add a `"polars-ops/semi_anti_join"` as well.

**Comment by Bidek56** [comment]
@alexander-beedie Does this code works when using more than one col in the USING clause? `USING(a,b)`? Thx

**Comment by Bidek56** [comment]
I just tried `polars  0.19.7`, this SQL works fine: `SELECT * FROM tbl_a LEFT ANTI JOIN tbl_b USING (b)`
But this SQL returns nothing, which is not correct. `SELECT * FROM tbl_a LEFT ANTI JOIN tbl_b USING (a,b)`
If the feature is not supported, maybe it should throw an error and note it in the docs? Thx

**Comment by alexander-beedie** [comment]
> @alexander-beedie Does this code works when using more than one col in the USING clause? `USING(a,b)`? Thx

Hmm, I'll take a look.

**Comment by alexander-beedie** [comment]
Ouch. Found out what's going on; it's not an issue with this PR - apparently we've had a long-standing bug in our `USING` syntax support where we just silently ate everything except the first column. Creating a PR to fix that now 😓

**Pull request merged**